composer.json

```diff
@@ -20,20 +20,22 @@
             "homepage": "https://github.com/friendsofsymfony/FOSHttpCache/contributors"
         }
     ],
+    "minimum-stability": "dev",
+    "prefer-stable": true,
     "require": {
-        "php": ">=5.4.8",
+        "php": ">=5.5.9",
         "symfony/event-dispatcher": "^2.3||^3.0",
         "symfony/options-resolver": "^2.3||^3.0",
-        "psr/http-message-implementation": "~1.0",
-        "php-http/adapter-implementation": "^0.1.0",
-        "php-http/discovery": "^0.1.1",
-        "php-http/message-decorator": "^0.1.0"
+        "php-http/client-implementation": "^1.0.0",
+        "php-http/discovery": "^0.6.0",
+        "php-http/message": "^0.2.1",
+        "puli/cli": "^1.0.0-beta8"
```

Comment:
Puli CLI collects configuration (bindings, available clients in our case) from all your installed composer packages and stores them under the .puli directory. This is done during composer install normally (by the composer plugin required by discovery) or of course you can do it manually by running `puli build`.

In order to do it, puli cli must be available somewhere. Either in your prefix as an executable, or by installing the puli cli package in your project or library.

To achieve full zero config, you have two options:

1. assume puli cli is installed by the user (either the phar or the package installed in the application where this library is used)
2. Install puli cli with this library which means puli cli will be made sure to be installed. However this way you don't let the user choose the way. If the puli CLI is already available, you essentially force the user to install the dependency anyway. I don't say it is bad, but I guess it is not how puli is imagined to work. That's why I said that for the long-term puli executable can be expected to be installed by the user, but for now, we can't expect this.